sparkly/writer.py

```diff
@@ -255,6 +275,188 @@ def write_partition_to_kafka(messages):
 
         rdd.mapPartitions(write_partition_to_kafka).count()
 
+    def redis(self,
+              key_by,
+              key_prefix=None,
+              key_delimiter='.',
+              group_by_key=False,
+              exclude_key_columns=False,
+              exclude_null_fields=False,
+              expire=None,
+              compression=None,
+              max_pipeline_size=100,
+              parallelism=None,
+              mode='overwrite',
+              host=None,
+              port=6379,
+              db=0,
+              redis_client_init=None):
+        """Write a dataframe to Redis as JSON.
+
+        Args:
+            key_by (list[str]): Column names that form the redis key for
+                each row. The columns are concatenated in the order they
+                appear.
+            key_prefix (str|None): Common prefix to add to all keys from
+                this DataFrame. Useful to namespace DataFrame exports.
+            key_delimiter (str|.): Characters to delimit different columns
+                while forming the key.
+            group_by_key (bool|False): If set, group rows that share the
+                same redis key together in an array before exporting. By
+                default if multiple rows share the same redis key, one
+                will overwrite the other.
+            exclude_key_columns (bool|False): If set, exclude all columns
+                that comprise the key from the value being exported to
+                redis.
+            exclude_null_fields (bool|False): If set, exclude all fields
+                of a row that are null from the value being exported to
+                redis.
+            expire (int|None): Expire the keys after this number of seconds.
+            compression (str|None): Compress each Redis entry using this
+                protocol. Currently bzip2, gzip and zlib are supported.
+            max_pipeline_size (int|100): Number of writes to pipeline.
+            parallelism (int|None): The max number of parallel tasks that
+                could be executed during the write stage
+                (see :ref:`controlling-the-load`).
+            mode (str|overwrite): Save mode. Options are:
+                 'overwrite': Overwrite existing data on a key by key
+                     basis.
+                 'ignore': Silently ignore if data already exists.
+                 'flush': Flush all existing data before writing.
+            host (str|None): Redis host. Either this or redis_client_init
+                must be provided. See below.
+            port (int|6379): Port redis is listening to.
+            db (int|0): Redis db to write to.
+            redis_client_init (callable|None): Bypass internal redis
```

Comment:
Yes, we need it, because redis-py sets up a client internally that it returns through `from_url` when you use it to parse `redis` and `rediss` URLs. So we're using it when resolving the URL. In this particular case it's the only extra thing that would end up in `options`, so I feel it's cleaner to have it here explicitly.